app/controllers/api/physical_servers_controller.rb

```diff
@@ -41,7 +41,8 @@ def restart_mgmt_controller_resource(type, id, _data)
     end
 
     def refresh_resource(type, id, _data = nil)
-      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" unless id
+      ensure_id_present(type, id)
+      ensure_resource_exists(type, id)
```

Comment:
I think it may be more appropriate to say:

```ruby
ensure_resource_exists(type, id) if single_request?
```

Since we're trying to express that we don't care if a resource exists at this point for a bulk operation, because it's handled elsewhere.